api/consumers.py

```diff
@@ -231,6 +233,25 @@ async def update_answers(self, event):
         results = await self.fetch_question_results(event["question_id"])
         await self.send(text_data=json.dumps({"type": "update_answers", "data": results}))
 
+    @database_sync_to_async
+    def add_to_duration_db(self, question_id, extension: int):
+        quiz_session_question = QuizSessionQuestion.objects.get(
+            quiz_session__code=self.code, question__id=question_id
+        )
+        quiz_session_question.extension += extension
+        quiz_session_question.save()
+        return quiz_session_question
+
+    async def add_to_duration(self, question_id, extension: int):
+        quiz_session_question = await self.add_to_duration_db(question_id, extension)
+        response = {
+            "type": "time_extended",
+            "question_opened_at": quiz_session_question.opened_at.isoformat(),
+            "extension": quiz_session_question.extension,
+        }
+        await self.channel_layer.group_send(f"quiz_session_{self.code}", response)
```

Comment:
@AustinfHunter, why do you think we need to send this message to the students' view?

Reply:
I'm assuming we'll need to notify student clients of additional time since the question locks after the set duration on the front-end. 

Without some sort of signal to the student client, there will be no way to update the total time before locking, essentially nullifying the extension.

Reply:
So, wouldn't it be better if the locking logic is controlled by the backend?

Student tries to submit answer; request goes to backend; if question is still unlocked accept the request; update UI.

Reply:
We could do that. Although I do feel like that removes the feedback associated with the frontend disabling answers for students once their time is up regardless of any attempt to submit an answer.

I can remove the message to the student clients for now and we can wait to see if any students notice the lack of visual feedback associated with a question timing out. If it does present a problem, we can always add the message back in fairly easily down the road.